### PR TITLE
Wc honeypot

### PR DESCRIPTION
**Pull request by ArtemAnoshin**

### CODE REVIEW

File: inc/cleantalk-settings.php [hunk 1]
```diff
@@ -258,6 +258,22 @@ function apbct_settings__set_fileds()
...
+                'forms__wc_honeypot'         => array(
+                    'title'           => __(
+                        'Add a hidden field to the order form as a trap for bots',
+                        'cleantalk-spam-protect'
+                    ),
+                    'description'     => __(
+                        'If the field is filled in during the order process, it means that the form was filled out by a bot, not a person.',
+                        'cleantalk-spam-protect'
+                    ),
+                    'reverse_trigger' => false,
```
Comment (safronik) [inline inc/cleantalk-settings.php:270]: If false, could be skipped.

File: inc/cleantalk-public-integrations.php [hunk 2]
```diff
@@ -464,18 +464,31 @@ function ct_woocommerce_checkout_check()
...
+        'sender_nickname' => $sender_nickname,
+        'post_info'       => $post_info,
+        'js_on'           => apbct_js_test('ct_checkjs', $_COOKIE, true),
+        'sender_info'     => array('sender_url' => null)
     );
 
+    /**
+     * Add honeypot_field to $base_call_data is forms__wc_honeypot on
+     */
+    if ( isset($apbct->settings['forms__wc_honeypot']) && $apbct->settings['forms__wc_honeypot'] ) {
```
Comment (safronik) [inline inc/cleantalk-public-integrations.php:479]: isset() is redundant. You could use 
`$apbct->settings['forms__wc_honeypot']`

File: inc/cleantalk-public-integrations.php [hunk 3]
```diff
@@ -464,18 +464,31 @@ function ct_woocommerce_checkout_check()
...
+        $honeypot_field = 1;
+
+        if ( isset($_POST['apbct_wc_honeypot']) && ! empty($_POST['apbct_wc_honeypot']) ) {
```
Comment (safronik) [inline inc/cleantalk-public-integrations.php:482]: These two conditions should be combined into one.

File: inc/cleantalk-common.php [hunk 4]
```diff
@@ -1092,6 +1092,17 @@ function apbct__styles_if_website_hidden()
 
         echo $styles;
     }
+
+    if ( isset($apbct->settings['forms__wc_honeypot']) && $apbct->settings['forms__wc_honeypot'] ) {
+        $styles = "
+		<style>
+		#apbct_wc_honeypot {
```
Comment (safronik) [inline inc/cleantalk-common.php:1099]: Could it be two of our honeypot fields on one page?
Isn't it better to use CSS class instead?

**Review by safronik — CHANGES_REQUESTED** [review]

File: inc/cleantalk-settings.php (hunk 1 above)
Comment (ArtemAnoshin) [reply inc/cleantalk-settings.php:270]: removed

File: inc/cleantalk-public-integrations.php (hunk 2 above)
Comment (ArtemAnoshin) [reply inc/cleantalk-public-integrations.php:479]: Fixed

File: inc/cleantalk-public-integrations.php (hunk 3 above)
Comment (ArtemAnoshin) [reply inc/cleantalk-public-integrations.php]: Fixed

File: inc/cleantalk-common.php (hunk 4 above)
Comment (ArtemAnoshin) [reply inc/cleantalk-common.php]: Fixed

**Review by Glomberg — APPROVED** [review]
Everything is ok.

**Review by safronik — APPROVED** [review]

**Pull request merged**